src/MoBi.Presentation/Tasks/ObservedDataTask.cs

```diff
@@ -278,6 +280,22 @@ public void RemoveResultsFromSimulations(IReadOnlyList<DataRepository> resultsTo
          }
       }
 
+      public void RemoveMultipleModules(IReadOnlyList<Module> modulesToRemove)
+      {
+         if (_dialogCreator.MessageBoxYesNo(AppConstants.Dialog.RemoveMultipleModules) != ViewResult.Yes)
+            return;
+
+         var macroCommand = new MoBiMacroCommand
+         {
+            Description = AppConstants.Commands.RemoveMultipleModules,
+            ObjectType = AppConstants.MoBiObjectTypes.Module,
+            CommandType = AppConstants.Commands.DeleteCommand
+         };
+
+         modulesToRemove.Each(x => macroCommand.Add(new RemoveModuleCommand(x)));
```

Comment:
When you move this code to `IInteractionTasksForModule`, you can use `GetRemoveCommand` to create the commands.

Reply:
hmm this is strange.
As now, I`m creating the command like this:

But I can see the "GetRemoveCommand" has 3 parameters:
GetRemoveCommand(Module objectToRemove, MoBiProject parent, IBuildingBlock buildingBlock)
2 of them which I do not have at this stage.
Checking the implementetion, I can see it is ignoring the 2 extra parameters:
public override IMoBiCommand GetRemoveCommand(Module objectToRemove, MoBiProject parent, IBuildingBlock buildingBlock) => new RemoveModuleCommand(objectToRemove);

Would it be ok to  use this setting null to the 2nd and 3rd parameter?

Let me know!

Reply:
Yes, you can use `null` in these cases. This is written this way in the inherited class `InteractionTasksForChildren<TParent, TChild, TEditTask>`. 

Let's change these two to optional parameters in that base class. There are other cases where they are not required also. And we won't need to keep using `null`

Reply:
> Let's change these two to optional parameters in that base class. There are other cases where they are not required also. And we won't need to keep using `null`

Actually, let's not do that right now - there's so much additional changes required that it will just make too much noise in the PR.

We can do that separately in another PR.